lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1673,6 +1674,52 @@ bool TypeChecker::typeCheckFunctionBodyUntil(FuncDecl *FD,
   BraceStmt *BS = FD->getBody();
   assert(BS && "Should have a body");
 
+  // FIXME: FIXME_NOW: DETERMINE: What is the appropriate place for this code to
+  //        live?
+  if (FD->hasSingleExpressionBody()) {
+    auto fnTypeLoc = FD->getBodyResultTypeLoc();
+    auto fnType = fnTypeLoc.getType();
```

Comment:
Nitpick: `fnType` is a confusing name for the result type.

Reply:
Fixed in 
https://github.com/apple/swift/commit/a310279dabda30f1de50b2a6bb2c11c0ef20868f where that variable is eliminated and the other, `fnTypeLoc` is renamed to `resultTypeLoc`.